test/integration/segmentation_camera_plugin.cc

```diff
@@ -0,0 +1,305 @@
+  box1->SetLocalPosition(rightPosition);
+  box1->SetLocalRotation(0, 0, 0);
+  box1->SetUserData("label", 1);
+  root->AddChild(box1);
+
+  // create box visual of different label
+  ignition::rendering::VisualPtr box2 = scene->CreateVisual();
+  box2->AddGeometry(scene->CreateBox());
+  box2->SetOrigin(0.0, 0.0, 0.0);
+  box2->SetLocalPosition(middlePosition);
+  box2->SetLocalRotation(0, 0, 0);
+  box2->SetUserData("label", 2);
+  root->AddChild(box2);
+}
+
+/////////////////////////////////////////////////
+void SegmentationCameraSensorTest::ImagesWithBuiltinSDF(
+  const std::string &_renderEngine)
+{
+  std::string path = ignition::common::joinPaths(PROJECT_SOURCE_PATH, "test",
+      "integration", "segmentation_camera_sensor_builtin.sdf");
+  sdf::SDFPtr doc(new sdf::SDF());
+  sdf::init(doc);
+  ASSERT_TRUE(sdf::readFile(path, doc));
+  ASSERT_NE(nullptr, doc->Root());
+  ASSERT_TRUE(doc->Root()->HasElement("model"));
+  auto modelPtr = doc->Root()->GetElement("model");
+  ASSERT_TRUE(modelPtr->HasElement("link"));
+  auto linkPtr = modelPtr->GetElement("link");
+  ASSERT_TRUE(linkPtr->HasElement("sensor"));
+  auto sensorPtr = linkPtr->GetElement("sensor");
+  ASSERT_TRUE(sensorPtr->HasElement("camera"));
+  auto cameraPtr = sensorPtr->GetElement("camera");
+  ASSERT_TRUE(cameraPtr->HasElement("image"));
+  auto imagePtr = cameraPtr->GetElement("image");
+
+  int width = imagePtr->Get<int>("width");
+  int height = imagePtr->Get<int>("height");
+
+  // Setup ign-rendering with an empty scene
+  auto *engine = ignition::rendering::engine(_renderEngine);
+  if (!engine)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' is not supported" << std::endl;
+    return;
+  }
+
+  ignition::rendering::ScenePtr scene = engine->CreateScene("scene");
+  BuildScene(scene);
+
+  ignition::sensors::Manager mgr;
+
+  sdf::Sensor sdfSensor;
+  sdfSensor.Load(sensorPtr);
+
+  std::string type = sdfSensor.TypeStr();
+  EXPECT_EQ(type, "segmentation_camera");
+
+  ignition::sensors::SegmentationCameraSensor *sensor =
+    mgr.CreateSensor<ignition::sensors::SegmentationCameraSensor>(sdfSensor);
+
+  EXPECT_NE(sensor, nullptr);
+  sensor->SetScene(scene);
+
+  EXPECT_EQ(width, sensor->ImageWidth());
+  EXPECT_EQ(height, sensor->ImageHeight());
+
+  auto camera = sensor->SegmentationCamera();
+  EXPECT_NE(camera, nullptr);
+
+  uint32_t backgroundLabel = 23;
+  camera->SetSegmentationType(SegmentationType::Semantic);
+  camera->EnableColoredMap(false);
+  camera->SetBackgroundLabel(backgroundLabel);
+  camera->SetLocalPosition(0.0, 0.0, 0.0);
+  camera->SetLocalRotation(0.0, 0.0, 0.0);
+
+  // Get the Msg
+  std::string topic =
+    "/test/integration/SegmentationCameraPlugin_imagesWithBuiltinSDF";
+  WaitForMessageTestHelper<ignition::msgs::Image> helper(topic);
+
+  // Update once to create image
+  mgr.RunOnce(std::chrono::steady_clock::duration::zero());
+
+  EXPECT_TRUE(helper.WaitForMessage()) << helper;
```

Comment:
It would also be good to check the camera info topic. Here's an example from the thermal camera: https://github.com/ignitionrobotics/ign-sensors/blob/a9517f8b7788838d373143fe87582ca02a8d7c89/test/integration/thermal_camera_plugin.cc#L192-L204